### PR TITLE
Hw-hb-05

### PR DESCRIPTION
**Pull request by RomanHreits**

### CODE REVIEW

File: src/main/java/com/cinema/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> query = session.createQuery("from User u "
+                    + "where u.email = :email", User.class);
+            query.setParameter("email", email);
+            return Optional.ofNullable(query.uniqueResult());
```
Comment (IrynaDim) [inline src/main/java/com/cinema/dao/impl/UserDaoImpl.java]: Why you don't catch exception here?

Comment (Sofasmile) [inline src/main/java/com/cinema/dao/impl/UserDaoImpl.java]:
```suggestion
            return query.uniqueResultOptional();
```

File: src/main/java/com/cinema/security/AuthenticationService.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.cinema.security;
+
+import com.cinema.exceptions.AuthenticationException;
+import com.cinema.model.User;
+
+public interface AuthenticationService {
+    User login(String email, String password) throws AuthenticationException;
+
+    User register(String email, String password) throws AuthenticationException;
```
Comment (Sofasmile) [inline src/main/java/com/cinema/security/AuthenticationService.java]:
```suggestion
    User register(String email, String password);
```

File: src/main/resources/log4j2.xml [hunk 3]
```diff
@@ -5,15 +5,14 @@
...
-        <RollingFile name="fileLogger" fileName="${basePath}/app-info.log" filePattern="${basePath}/app-info-%d{yyyy-MM-dd}.log">
-            <PatternLayout>
-                <pattern>[%-5level] %d{yyyy-MM-dd HH:mm:ss.SSS} [%t] %c{1} - %msg%n</pattern>
-            </PatternLayout>
-            <Policies>
-                <TimeBasedTriggeringPolicy interval="1" modulate="true" />
-            </Policies>
-        </RollingFile>
-
+<!--        <RollingFile name="fileLogger" fileName="${basePath}/app-info.log" filePattern="${basePath}/app-info-%d{yyyy-MM-dd}.log">-->
```
Comment (Sofasmile) [inline src/main/resources/log4j2.xml]: let's remove commented code from repository

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**